src/basic/pidfd-util.c

```diff
@@ -33,6 +34,34 @@ static int pidfd_check_pidfs(void) {
         return (is_pidfs = fd_is_fs_type(fd, PID_FS_MAGIC));
 }
 
+int pidfd_get_namespace(int fd, unsigned long ns_type_flag) {
+        static bool cached_supported = true;
+
+        assert(fd >= 0);
+
+        /* If we know ahead of time that pidfs is unavailable, shortcut things. But otherwise we don't
+         * call pidfd_check_pidfs() here, which is kinda extraneous and our own cache is required
+         * anyways (pidfs is introduced in kernel 6.9 while ioctl support there is added in 6.11). */
+        if (is_pidfs == 0 || !cached_supported)
+                return -EOPNOTSUPP;
+
+        int nsfd = ioctl(fd, ns_type_flag);
```

Comment:
hmm, ns_type_flag is an ioctl command number? shouldn't be called "flag" then.

reading through this PR it's not clear to me with what kind of 2nd params this function shall be called, this deserves a comment.

(i see now this is mentioned in the commit msg, but this should be in the comment too)